nntrainer/src/tensor.cpp

```diff
@@ -783,17 +783,25 @@ Tensor Tensor::apply(std::function<float(float)> f) const {
 Tensor Tensor::apply(std::function<Tensor(Tensor)> f) const { return f(*this); }
 
 void Tensor::print(std::ostream &out) const {
-  unsigned int i, j, k, l;
-  std::stringstream ss;
-
+  out << "<Tensor Object at " << this << ">" << std::endl;
   const float *data = getData();
-  for (k = 0; k < dim.batch(); k++) {
-    for (l = 0; l < dim.channel(); l++) {
-      for (i = 0; i < dim.height(); i++) {
-        for (j = 0; j < dim.width(); j++) {
-          out << data[k * dim.getFeatureLen() + l * dim.width() * dim.height() +
-                      i * dim.width() + j]
-              << " ";
+
+  unsigned int len = length();
+
+  out << dim;
+
+  if (len > 100) {
+    out << '[' << data[0] << ' ' << data[1] << ' ' << data[2] << " ... "
```

Comment:
How about skip printing data in the model summary. If someone needs data, they can directly inquire tensor.